### PR TITLE
fix: decode _xHHHH_ escapes when reading inline string cells

### PR DESCRIPTION
**Pull request by nkuprins**
Related: #696 

## Purpose of the pull request

Characters that XML 1.0 forbids are stored in a cell as `_xHHHH_` escapes. Fesod undid them only for cells backed by `sharedStrings.xml`, so `t="inlineStr"` and `t="str"` reached the caller with the raw escape. A value written by Fesod did not survive being read back by Fesod. The clearest case is a value escaped with `EscapeHexCellWriteHandler`, which stores the literal `_xB9f0_` as `_x005F_xB9f0_` precisely so a decoding reader restores it:

| reading a cell written as `Product_x005F_xB9f0_Code` | result |
| --- | --- |
| POI | `Product_xB9f0_Code` |
| Fesod, before / after | `Product_x005F_xB9f0_Code` / `Product_xB9f0_Code` |

Why this is a defect and not intended behaviour:
- Decoding is already an asserted contract for the other storage form - `CompatibilityTest#readXlsxWithEscapeSequence` pins it for `sharedStrings.xml`.
- POI decodes inline strings on both of its read paths - the DOM `XSSFCell` and the streaming `XSSFSheetXMLHandler`, each via `XSSFRichTextString#getString()`.
- No test asserts that a raw escape survives a read.

## What's changed?

- **`XlsxEscapeUtils` (new)** - `utfDecode` moved here from `SharedStringsTableHandler` unchanged, and both read paths now call it. The smaller change would have been to make the existing method `public` and call it from `CellTagHandler`, but that leaves a cell parser depending on the `sharedStrings.xml` parser for decoding that has nothing to do with shared strings, and the next caller inherits the same detour. The escape convention belongs to neither handler, so it moved to `util/`. Happy to switch to the two-line version if you prefer the smaller diff.
- **`CellTagHandler`** - the `DIRECT_STRING` branch now decodes instead of falling through to the `ERROR` branch; it sets the same `STRING` type as before.
- **`HexEscapeRoundTripTest` (new)** - one round trip through the handler: `Product_x0002_Code` written with `EscapeHexCellWriteHandler` is stored as `Product_x005F_x0002_Code` and must read back as the literal `Product_x0002_Code`.

## Checklist

- [x] I have read the [Contributor Guide](https://fesod.apache.org/community/contribution/).
- [x] I have written the necessary doc or comment.
- [x] I have added the necessary unit tests and all cases have passed.

### CODE REVIEW

File: fesod-sheet/src/main/java/org/apache/fesod/sheet/analysis/v07/handlers/CellTagHandler.java [hunk 1]
```diff
@@ -116,6 +117,10 @@ public void endElement(XlsxReadContext xlsxReadContext, String name) {
                 tempCellData.setStringValue(stringValue);
                 break;
             case DIRECT_STRING:
+                // Undo the '_xHHHH_' escapes of characters XML forbids
```
Comment (alaahong) [inline fesod-sheet/src/main/java/org/apache/fesod/sheet/analysis/v07/handlers/CellTagHandler.java:110]: The reader now decodes `_xHHHH_`, but the default writer (SXSSF inline strings) never escapes them. Verified locally: writing `Product_x0002_Code` with the default writer and reading it back now returns `Product\u0002Code` (STX control char) instead of the literal — before this change it returned the literal. For third-party files this is the correct fix for #696, but for fesod→fesod round trips it's a silent behaviour change. Suggest aligning the writer (escape by default, like `EscapeHexCellWriteHandler`) or documenting that literal `_xHHHH_` text requires registering that handler.

File: fesod-sheet/src/test/java/org/apache/fesod/sheet/readwrite/HexEscapeRoundTripTest.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        data.setName("Product_x0002_Code");
+        File file = createTempFile("hex-escape", ExcelFormat.XLSX);
+        FesodSheet.write(file, SimpleData.class).sheet().doWrite(Collections.singletonList(data));
+
+        CollectingReadListener<SimpleData> listener = new CollectingReadListener<>();
+        FesodSheet.read(file, SimpleData.class, listener).sheet().doRead();
+        List<SimpleData> rows = listener.getRows();
+
+        Assertions.assertEquals(1, rows.size());
+        Assertions.assertEquals("Product" + STX + "Code", rows.get(0).getName());
```
Comment (alaahong) [inline fesod-sheet/src/test/java/org/apache/fesod/sheet/readwrite/HexEscapeRoundTripTest.java]: This assertion only holds because the default writer does not escape `_xHHHH_`. If the writer is fixed to escape (spec-correct), the same input reads back as the literal `Product_x0002_Code` and this test would fail for the wrong reason. Suggest writing via `EscapeHexCellWriteHandler` (as in #1001's round-trip test) or using a POI/Excel-generated fixture containing real escapes.

File: fesod-sheet/src/main/java/org/apache/fesod/sheet/util/XlsxEscapeUtils.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+package org.apache.fesod.sheet.util;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Escapes defined by the Office Open XML spec, section 3.18.9.
+ *
+ *
+ */
```
Comment (alaahong) [inline fesod-sheet/src/main/java/org/apache/fesod/sheet/util/XlsxEscapeUtils.java:45]: Since `utfDecode` is now a shared public API, consider adding a parameterized unit test (edge cases: adjacent escapes, `_x000D_`, uppercase `_X`, truncated sequences, non-hex digits, pattern at end of string) and expanding the class-level javadoc (purpose, the two call sites, OOXML §3.18.9).

**Review by alaahong — COMMENTED** [review]
Nice cleanup — moving `utfDecode` to `util/` so `CellTagHandler` doesn't depend on the shared-strings parser is the right call, and `t="str"` is covered since it maps to `DIRECT_STRING`. Comments inline. One more note: the escape convention is now implemented twice (`XlsxEscapeUtils` for reading, `EscapeHexCellWriteHandler` for writing); consider cross-referencing them or unifying in `XlsxEscapeUtils` to prevent silent drift.

File: fesod-sheet/src/main/java/org/apache/fesod/sheet/analysis/v07/handlers/CellTagHandler.java (hunk 1 above)
Comment (nkuprins) [reply fesod-sheet/src/main/java/org/apache/fesod/sheet/analysis/v07/handlers/CellTagHandler.java:110]: Hi @alaahong,

I see, but if I take the first option (escape by default), then do you agree that we need an opt-out - `escapeHexText(false)` on the write? Without it, escapes that arrived from another producer get escaped again on the way in, with no way to write them through unchanged. 

Also, probably escape by default isn't a small addition to this PR, so not sure if I should do it here.

Thank you for the review!

Comment (nkuprins) [reply fesod-sheet/src/main/java/org/apache/fesod/sheet/analysis/v07/handlers/CellTagHandler.java:110]: I chose to add javadoc note instead, as suggested in the second option. This probably belongs on the website too, but I'm not sure which section - FAQ?

**Review by alaahong — APPROVED** [review]
LGTM

File: fesod-sheet/src/test/java/org/apache/fesod/sheet/util/XlsxEscapeUtilsTest.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+     * POI is the reference for this decoding, so it is also the oracle: any input where the two disagree is a defect
+     * here, whatever the table above says.
+     */
+    @ParameterizedTest(name = "[{index}] {0} decodes as POI does")
+    @MethodSource
+    void matchesPoiForEveryInput(String input) {
+        Assertions.assertEquals(new XSSFRichTextString(input).getString(), XlsxEscapeUtils.utfDecode(input));
+    }
+
+    static Stream<String> matchesPoiForEveryInput() {
```
[the remark this answers is not in the document]
Comment (nkuprins) [reply fesod-sheet/src/test/java/org/apache/fesod/sheet/util/XlsxEscapeUtilsTest.java:102]: The source is the union of `decodesEscapes()` and `leavesTextWithoutAnEscapeAlone()`, so "every
input" means every input in this class

File: fesod-sheet/src/test/java/org/apache/fesod/sheet/readwrite/HexEscapeRoundTripTest.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+                .registerWriteHandler(new EscapeHexCellWriteHandler())
+                .sheet()
+                .doWrite(Collections.singletonList(data));
+
+        CollectingReadListener<SimpleData> listener = new CollectingReadListener<>();
+        FesodSheet.read(file, SimpleData.class, listener).sheet().doRead();
+        List<SimpleData> rows = listener.getRows();
+
+        Assertions.assertEquals(1, rows.size());
+        Assertions.assertEquals("Product_x0002_Code", rows.get(0).getName());
```
[the remark this answers is not in the document]
Comment (nkuprins) [reply fesod-sheet/src/test/java/org/apache/fesod/sheet/readwrite/HexEscapeRoundTripTest.java:65]: PR description changed

File: fesod-sheet/src/test/java/org/apache/fesod/sheet/util/XlsxEscapeUtilsTest.java [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+                Arguments.of("_x0041_", "A"),
+                Arguments.of("_x0041_tail", "Atail"),
+                Arguments.of("head_x0041_", "headA"),
+                Arguments.of("head_x0041_tail", "headAtail"),
+                Arguments.of("_x0041__x0042_", "AB"),
+                Arguments.of("__x0041_", "_A"),
+                Arguments.of("_x000D_", "\r"),
+                // the hex digits are case insensitive, the leading x is not - see leavesTextWithoutAnEscapeAlone
+                Arguments.of("_x00e9_", "é"),
+                Arguments.of("_x00E9_", "é"),
```
[the remark this answers is not in the document]
Comment (nkuprins) [reply fesod-sheet/src/test/java/org/apache/fesod/sheet/util/XlsxEscapeUtilsTest.java:51]: Doesn't seem to be a problem - non-ASCII literals are already used in `CompatibilityTest`, etc.
